pkg/oauth/apiserver/oauth_apiserver.go

```diff
@@ -40,8 +40,8 @@ type OAuthServerConfig struct {
 	// KubeClient is kubeclient with enough permission for the auth API
 	KubeClient kclientset.Interface
 
-	// OpenShiftClient is osclient with enough permission for the auth API
-	OpenShiftClient osclient.Interface
+	// RouteClient provides a client for OpenShift routes API.
+	RouteClient routeclient.Interface
```

Comment:
Same `routeclient.RouteInterface`, see below how we use specific interfaces as well.